taichi/transforms/scalarize.cpp

```diff
@@ -45,9 +45,12 @@ class Scalarize : public IRVisitor {
       // Needs scalarize
       auto dest_tensor_type = dest_dtype->template as<TensorType>();
       auto val_tensor_type = val_dtype->template as<TensorType>();
+
       TI_ASSERT(dest_tensor_type->get_shape() == val_tensor_type->get_shape());
-      TI_ASSERT(dest_tensor_type->get_element_type() ==
-                val_tensor_type->get_element_type());
+      // For sqrt/exp/log with int-type operand, we automatically set the
```

Comment:
IMO this is normal and it is handled by https://github.com/taichi-dev/taichi/blob/18c946ea67d23265896d3fb967482dfdd4e63795/taichi/transforms/type_check.cpp#L21-L41.

Reply:
Yup, it's actually fine for `compile_kernel()` since `type_check()` is called immediately after `lower_ast()`. However, it doesn't make much sense to me as `type_check()` for `compile_function()` is placed at the very end:

![2022-09-20 09-38-34 的屏幕截图](https://user-images.githubusercontent.com/22334008/191148535-8d17dd8b-a32f-489d-a579-1c5b9fa9ba7b.png)

Reply:
Ah here I mean `load_store_forwarding()` is unrelated, which should not appear in the comments.